### ISSUE TITLE
[HOLD for payment 2024-12-19] [Search v1.2] - Different approval behavior when approving one held expense vs multiple held expenses

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53012]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.66-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N/A - new feature, doesn't exist in prod
**Email or phone of affected tester (no customers):** applausetester+kh0911001@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Add approvals feature is enabled.
- Admin is the approver.

1. Go to staging.new.expensify.com
2. Go to workspace chat.
3. Submit an expense and hold the expense.
4. Go to Search.
5. Click Approve button on the expense.
6. Note that it opens expense RHP and the approval can only be done from the expense RHP.
7. Go back to workspace chat.
8. Submit another expense (so there are two expenses in the same report, with one held and the other unheld).
9. Go to Search > Outstanding.
10. Click Approve.

## Expected Result:
App should open expense report RHP so that approval can be done by approving the pending amount or all amount.

## Actual Result:
Expense report with held expense can be approved without the confirmation modal in Search.

There is inconsistency when approving only one held expense and approving two expenses with different held status.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/91db1fc1-82ea-4f9b-b554-bffe73002ef5

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by mountiny** [comment]
cc @luacmartins can you have a look please

I dont think this is a blocker as the feature works its just a different UX to discuss

**Comment by luacmartins** [comment]
I agree it doesn't need to be a blocker. I'll take a look at it since [my PR](https://github.com/Expensify/App/pull/52441) introduced this feature

**Comment by trjExpensify** [comment]
Just so I'm following along:

- If you click `Approve` on a single-expense report when the expense is held, it opens the RHP. 
- If you click `Approve` on a multi-expense report when one of the expenses is held, it approves the report not opens the RHP. 

So the second one is the bug, right?

**Comment by luacmartins** [comment]
Yea, that second one should also open the RHP

**Comment by luacmartins** [comment]
Working on it this week

**Comment by luacmartins** [comment]
PR in review

**Comment by trjExpensify** [comment]
Looks like @rushatgabhane needs to be compensated for the review of the PR, right? https://github.com/Expensify/App/pull/53613

Can we get the checklist please :)

**Comment by luacmartins** [comment]
>Looks like @rushatgabhane needs to be compensated for the review of the PR, right? https://github.com/Expensify/App/pull/53613

Yes, just waiting on checklist and payment

**Comment by trjExpensify** [comment]
👋 @rushatgabhane checklist time, please. Thanks!

**Comment by trjExpensify** [comment]
Same melv, awaiting the checklist.

**Comment by rushatgabhane** [comment]
1. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/52441

2. If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:  Not critical. just a consistency issue


3. If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again. N.A.

**Comment by trjExpensify** [comment]
>If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again. N.A.

Bit of a bug bear of mine, but how is "not applicable" a viable response to the question here? Please do expand on the reason why you don't think we need a regression test for this particular issue.

**Comment by trjExpensify** [comment]
Payment summary as follows: 

- $250 to @rushatgabhane for the C+ review

**Comment by rushatgabhane** [comment]
updated the regression test response - https://github.com/Expensify/App/issues/53012#issuecomment-2580205581

**Comment by luacmartins** [comment]
>@luacmartins has a regression test been added for this feature elsewhere?

I'm not sure if we're covering this in TCs yet. @IuliiaHerets can you confirm if this bug report came from a TC or just exploration?

**Comment by trjExpensify** [comment]
CC: @kavimuru @isagoico @mvtglobally as well on that Q!

**Comment by IuliiaHerets** [comment]
@luacmartins Tester confirmed that find this issue by eploratory, not follow some TR step or PR

**Comment by luacmartins** [comment]
Thanks! @IuliiaHerets can we add the OP test steps to TR please?

**Comment by trjExpensify** [comment]
@luliiaHerets any progress on that?

**Comment by trjExpensify** [comment]
I've added them to a GH for applause, they can confirm if they've been added there and close, or add them. 

We're done here!

**Comment by garrettmknight** [comment]
$250 approved for @rushatgabhane
